zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,875 @@
+                ),
+            )
+
+    # Check if this is a single column out of a dataset.
+    if bz.ndim(expr) != 1:
+        raise TypeError(
+            'expression was not tabular or array-like,'
+            ' %s dimensions: %d' % (
+                'too many' if bz.ndim(expr) > 1 else 'not enough',
+                bz.ndim(expr),
+            ),
+        )
+
+    single_column = None
+    if isscalar(expr.dshape.measure):
+        # This is a single column. Record which column we are to return
+        # but create the entire dataset.
+        single_column = rename = expr._name
+        field_hit = False
+        if not isinstance(expr, traversable_nodes):
+            raise TypeError(
+                "expression '%s' was array-like but not a simple field of"
+                " some larger table" % str(expr),
+            )
+        while isinstance(expr, traversable_nodes):
+            if isinstance(expr, bz.expr.Field):
+                if not field_hit:
+                    field_hit = True
+                else:
+                    break
+            rename = expr._name
+            expr = expr._child
+        expr = expr.relabel({rename: single_column})
+
+    measure = expr.dshape.measure
+    if not isrecord(measure) or AD_FIELD_NAME not in measure.names:
+        raise TypeError(
+            "expr must be a collection of records with at least an '{ad}'"
+            " field. Fields provided: '{fields}'\nhint: maybe you need to use "
+            " `relabel` to change your field names".format(
+                ad=AD_FIELD_NAME,
+                fields=measure,
+            ),
+        )
+    _check_datetime_field(AD_FIELD_NAME, measure)
+
+    if TS_FIELD_NAME not in measure.names:
+        expr = bz.transform(expr, **{TS_FIELD_NAME: expr[AD_FIELD_NAME]})
+        if deltas is not None:
+            deltas = bz.transform(
+                deltas,
+                **{TS_FIELD_NAME: deltas[AD_FIELD_NAME]}
+            )
+    else:
+        _check_datetime_field(TS_FIELD_NAME, measure)
+
+    if deltas is not None and (sorted(deltas.dshape.measure.fields) !=
+                               sorted(measure.fields)):
+        raise TypeError(
+            'baseline measure != deltas measure:\n%s != %s' % (
+                measure,
+                deltas.dshape.measure,
+            ),
+        )
+
+    # Ensure that we have a data resource to execute the query against.
+    _check_resources('expr', expr, resources)
+    _check_resources('deltas', deltas, resources)
+
+    # Create or retrieve the Pipeline API dataset.
+    ds = new_dataset(expr, deltas)
+    # Register our new dataset with the loader.
+    (loader if loader is not None else global_loader)[ds] = ExprData(
+        expr,
+        deltas,
+        resources,
+    )
+    if single_column is not None:
+        # We were passed a single column, extract and return it.
+        return getattr(ds, single_column)
+    return ds
```

Comment:
This function does fail if a single column is incorrect. It will create the dataset correctly but then fail when the column is requested:

```python
In [16]: ds = bz.Data([{'asof_date': dt, 'a': {'b': 1}}])

In [17]: from_blaze(ds)
/home/yui/quantopian/research_env/base/src/qexec_repo/zipline_repo/zipline/pipeline/loaders/blazerred from expr: _5
  warnings.warn(NoDeltasWarning(expr))
Out[17]: <DataSet: 'BlazeDataSet_0'>

In [18]: from_blaze(ds).a
/home/yui/quantopian/research_env/base/src/qexec_repo/zipline_repo/zipline/pipeline/loaders/blazerred from expr: _5
  warnings.warn(NoDeltasWarning(expr))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-dbcf18c0e16a> in <module>()
----> 1 from_blaze(ds).a

/home/yui/quantopian/research_env/base/src/qexec_repo/zipline_repo/zipline/pipeline/loaders/blaze
    245     def __get__(self, instance, owner):
    246         raise AttributeError(
--> 247             self.error_format.format(field=self._field, type_=self._type),
    248         )
    249 

AttributeError: field 'a' was a non numpy compatible type: '{b: int64}'

```